File: Active/TypeScript/src/P007/P007.ts
// Project Euler
// Problem 7: The 10001st prime number

import isPrime from "../lib/isPrime";

export function compute007(nth: number): number {
    let primeTh: number = 0;
    let i = 1

    for (; primeTh !== nth; i++) {
        if (isPrime(i)) {
            primeTh += 1;
        }
    }

    // minus 1 because, i is getting added before the for loop stop
    return (i - 1);
}

function run007(): void {
    const nth: number = 10001;
    console.log(compute007(nth).toString());
}

run007();

File: Active/Go/solutions/p007.go
// Project Euler
// Problem 7: The 10_001st prime number

package go_euler

import "go_euler/lib"

func Problem_007(limit int) int {
	prime_th := 0

	i := 1
	for ; prime_th != limit; i++ {
		if go_euler.IsPrime(i) {
			prime_th += 1
		}
	}

    // minus 1 because, i is getting added before the for loop stop
    return (i - 1);
}
